pkg/querier/tripperware/query.go

```diff
@@ -721,3 +732,33 @@ func marshalHistogramBucket(b HistogramBucket, stream *jsoniter.Stream) {
 	jsonutil.MarshalFloat(b.Count, stream)
 	stream.WriteArrayEnd()
 }
+
+func (s *PrometheusResponseStats) MarshalJSON() ([]byte, error) {
+	stats := struct {
+		Samples *PrometheusResponseSamplesStats `json:"samples"`
+	}{
+		Samples: s.Samples,
+	}
+	if s.Samples.TotalQueryableSamplesPerStep == nil {
+		s.Samples.TotalQueryableSamplesPerStep = []*PrometheusResponseQueryableSamplesStatsPerStep{}
+	}
+	return json.Marshal(stats)
+}
+
+func SetRequestHeaders(h http.Header, defaultCodec string, compression string) {
+	if compression == GzipCompression {
+		h.Set("Accept-Encoding", GzipCompression)
```

Comment:
Do we support Snappy as well?

Reply:
not anymore since that was added with the protobuf handler I created. it is not in the Prometheus compression handler. I plan to create a PR there to add it in the future, but for now no compression is actually a good option since protobuf responses are smaller than json anyway. The default is still gzip.